tests/test_projectgen.py

```diff
@@ -2470,6 +2470,260 @@ def test_relation_strength_mssql(self):
+        importer.tool = DbIliMode.ili2pg
+        importer.configuration = iliimporter_config(importer.tool)
+        importer.configuration.ilifile = testdata_path(
+            "ilimodels//OneToOneRelations.ili"
+        )
+        importer.configuration.ilimodels = "GeolTest"
+        importer.configuration.dbschema = "cardinality_max_{:%Y%m%d%H%M%S%f}".format(
+            datetime.datetime.now()
+        )
+        importer.configuration.srs_code = 2056
+        importer.configuration.inheritance = "smart2"
+        importer.stdout.connect(self.print_info)
+        importer.stderr.connect(self.print_error)
+        assert importer.run() == iliimporter.Importer.SUCCESS
+
+        generator = Generator(
+            DbIliMode.ili2pg,
+            get_pg_connection_string(),
+            importer.configuration.inheritance,
+            importer.configuration.dbschema,
+        )
+
+        available_layers = generator.layers()
+        relations, _ = generator.relations(available_layers)
+        legend = generator.legend(available_layers)
+
+        project = Project()
+        project.layers = available_layers
+        project.relations = relations
+        project.legend = legend
+        project.post_generate()
+
+        qgis_project = QgsProject.instance()
+        project.create(None, qgis_project)
+
+        contact_layer = None
+        for layer in available_layers:
+            if layer.name == "contact":
+                contact_layer = layer
+                break
+        assert contact_layer
+
+        tab_address = None
+        tab_identificator = None
+        tab_ahvnr = None
+        tab_job = None
+        efc = contact_layer.layer.editFormConfig()
+        for tab in efc.tabs():
+            if tab.name() == "address":
+                tab_address = tab
+            elif tab.name() == "identificator":
+                tab_identificator = tab
+            elif tab.name() == "ahvnr":
+                tab_ahvnr = tab
+            elif tab.name() == "job":
+                tab_job = tab
+        assert tab_address
+        assert tab_identificator
+        assert tab_ahvnr
+        assert tab_job
+
+        if Qgis.QGIS_VERSION_INT < 31800:
+            return
+
+        for child in tab_address.children():
+            self.assertIn("one_to_one", child.relationEditorConfiguration())
+            self.assertTrue(child.relationEditorConfiguration()["one_to_one"])
+
+        for child in tab_identificator.children():
+            self.assertNotIn("one_to_one", child.relationEditorConfiguration())
+
+        for child in tab_ahvnr.children():
+            self.assertIn("one_to_one", child.relationEditorConfiguration())
+            self.assertTrue(child.relationEditorConfiguration()["one_to_one"])
+
+        for child in tab_job.children():
+            self.assertIn("one_to_one", child.relationEditorConfiguration())
+            self.assertTrue(child.relationEditorConfiguration()["one_to_one"])
```

Comment:
Shouldn't the job be not one to one, since the cardinalityMax to job is * (instead of 1)?

Reply:
mmh I think you are right. A contact can have more jobs, and a job can have only one contact 

![image](https://user-images.githubusercontent.com/9881900/191928608-f14c5d03-2b28-4b4e-a8d1-ced3d812f219.png)

Reply:
Yes. I'm a little confused. Is not the cardinalityMax for this relation == *?

Reply:
I am confused too :) This is what I get in relations_info:
```
referencing_layer: job
referenced_layer: contact
cardinality_max: 1
```
For this model
```
    ASSOCIATION Contact_Job =
        Contact -- {0..1} Contact;
        Job -- {0..*} Job;
    END Contact_Job;
```

Are we maybe just using the wrong "direction" of the relation? I mean shold we look for `Contact_Job.Job` instead of `Contact_Job.Contact`?

Reply:
Yes. It should be `Contact_Job.Job` -> The direction to the child items.